src/Explorer/Panes/DeleteDatabaseConfirmationPanel.tsx

```diff
@@ -0,0 +1,187 @@
+    };
+  };
+
+  const submit = async (): Promise<void> => {
+    const { selectedDatabase, explorer } = props;
+    if (!isValid()) {
+      setFormError("Input database name does not match the selected database");
+      NotificationConsoleUtils.logConsoleMessage(
+        ConsoleDataType.Error,
+        `Error while deleting collection ${selectedDatabase && selectedDatabase.id()}`
+      );
+      return;
+    }
+    setFormError("");
+    setLoadingTrue();
+
+    const startKey: number = TelemetryProcessor.traceStart(Action.DeleteDatabase, {
+      databaseId: selectedDatabase.id(),
+      dataExplorerArea: Constants.Areas.ContextualPane,
+      paneTitle: "Delete Database",
+    });
+
+    try {
+      await deleteDatabase(selectedDatabase.id());
+      setLoadingFalse();
+      props.closePanel();
+      explorer.refreshAllDatabases();
+      explorer.tabsManager.closeTabsByComparator((tab) => tab.node?.id() === selectedDatabase.id());
+      explorer.selectedNode(undefined);
+      selectedDatabase
+        .collections()
+        .forEach((collection: ViewModels.Collection) =>
+          explorer.tabsManager.closeTabsByComparator(
+            (tab) =>
+              tab.node?.id() === collection.id() &&
+              (tab.node as ViewModels.Collection).databaseId === collection.databaseId
+          )
+        );
+      TelemetryProcessor.traceSuccess(
+        Action.DeleteDatabase,
+        {
+          databaseId: selectedDatabase.id(),
+          dataExplorerArea: Constants.Areas.ContextualPane,
+          paneTitle: "Delete Database",
+        },
+        startKey
+      );
+
+      if (shouldRecordFeedback()) {
+        const deleteFeedback = new DeleteFeedback(
+          explorer.databaseAccount().id,
+          explorer.databaseAccount().name,
+          DefaultExperienceUtility.getApiKindFromDefaultExperience(explorer.defaultExperience()),
+          databaseFeedbackInput
+        );
+
+        TelemetryProcessor.trace(Action.DeleteDatabase, ActionModifiers.Mark, {
+          message: JSON.stringify(deleteFeedback, Object.getOwnPropertyNames(deleteFeedback)),
+        });
+        setDatabaseFeedbackInput("");
+      }
+    } catch (error) {
+      setLoadingFalse();
+      setFormError(error);
+      const errorMessage = getErrorMessage(error);
+      TelemetryProcessor.traceFailure(
+        Action.DeleteDatabase,
+        {
+          databaseId: selectedDatabase.id(),
+          dataExplorerArea: Constants.Areas.ContextualPane,
+          paneTitle: "Delete Database",
+          error: errorMessage,
+          errorStack: getErrorStack(error),
+        },
+        startKey
+      );
+    }
+  };
+
+  const shouldRecordFeedback = (): boolean => {
+    const { explorer } = props;
+    return explorer.isLastNonEmptyDatabase() || (explorer.isLastDatabase() && explorer.isSelectedDatabaseShared());
+  };
+
+  const isValid = (): boolean => {
```

Comment:
We should be able to do this check inline:

`if (!selectedDatabase?.id() || databaseInput !== selectedDatabase.id()) {
	...
}`

Reply:
Updated code.

Reply:
The condition that I wrote is for checking whether the input is invalid, which is what we are doing in the `submit` function (see line 55 of DeleteDatabaseConfirmationPanel.tsx).

What I meant was that we don't really need this function. We can just do this check inline (in line 55).

Reply:
Changes updated. Thank you.